src/BatteryStateOfChargeService.h

```diff
@@ -42,8 +42,17 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     void addData(const BatteryData& batteryData);
 
 private:
-    // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
 
-    // Recommend adding necessary variables here
+    QTime previousTime_;
+    QTime return_time;
+    QTime zero_time;
```

Comment:
I would not recommend using a private member for this. I'd actually recommend using just a zero constructor `QTime(0, 0)` whenever you need a `zero_time`. This is because `QTime(0, 0)` is just as understandable as `zero_time` and it'll make for less code that needs to be read since you don't need a private member for it.